test/plugins/individual-collateral/flux-finance/FTokenFiatCollateral.test.ts

```diff
@@ -0,0 +1,301 @@
+
+      const collateral = await deployCollateral(collateralOpts)
+      const erc20 = await ethers.getContractAt('ICToken', collateralOpts.erc20 as string) // the fToken
+      const tokDecimals = await erc20.decimals()
+
+      return {
+        alice,
+        collateral,
+        chainlinkFeed,
+        tok: erc20,
+        tokDecimals,
+      }
+    }
+
+    return makeCollateralFixtureContext
+  }
+
+  const deployCollateralMockContext = async (
+    opts: FTokenCollateralOpts = {}
+  ): Promise<CollateralFixtureContext> => {
+    const collateralOpts = { ...defaultCollateralOpts, ...opts }
+
+    const MockV3AggregatorFactory = <MockV3Aggregator__factory>(
+      await ethers.getContractFactory('MockV3Aggregator')
+    )
+
+    const chainlinkFeed = <MockV3Aggregator>await MockV3AggregatorFactory.deploy(6, bn('1e6'))
+    collateralOpts.chainlinkFeed = chainlinkFeed.address
+
+    const FTokenMockFactory = await ethers.getContractFactory('CTokenMock')
+    const erc20 = await FTokenMockFactory.deploy('Mock FToken', 'Mock Ftk', curr.underlying)
+    collateralOpts.erc20 = erc20.address
+
+    const collateral = await deployCollateral(collateralOpts)
+    const tokDecimals = await erc20.decimals()
+
+    return {
+      collateral,
+      chainlinkFeed,
+      tok: erc20,
+      tokDecimals,
+    }
+  }
+
+  /*
+  Define helper functions
+*/
+
+  const mintCollateralTo: MintCollateralFunc<CollateralFixtureContext> = async (
+    ctx: CollateralFixtureContext,
+    amount: BigNumberish,
+    user: SignerWithAddress,
+    recipient: string
+  ) => {
+    const tok = ctx.tok as ICToken
+    const underlying = await ethers.getContractAt('IERC20Metadata', await tok.underlying())
+    await mintFToken(underlying, curr.holderUnderlying, tok, amount, recipient)
+  }
+
+  const appreciateRefPerTok = async (ctx: CollateralFixtureContext) => {
+    await advanceBlocks(1)
+    await (ctx.tok as ICToken).exchangeRateCurrent()
+  }
+
+  const reduceRefPerTok = async () => {
+    return
+  }
+
+  const collateralSpecificConstructorTests = () => {
+    it('Should validate comptroller arg', async () => {
+      await expect(deployCollateral({ comptroller: ZERO_ADDRESS })).to.be.revertedWith(
+        'comptroller missing'
+      )
+    })
+  }
+
+  const collateralSpecificStatusTests = () => {
+    it('enters DISABLED state if refPerTok falls', async () => {
+      const { collateral, tok } = await deployCollateralMockContext()
+      const before = await collateral.refPerTok()
+      expect(await collateral.status()).to.equal(CollateralStatus.SOUND)
+      expect(before).to.be.gt(0)
+      await (tok as CTokenMock).setExchangeRate(before.sub(1))
+      await collateral.refresh()
+      const after = await collateral.refPerTok()
+      expect(before).to.be.gt(after)
+      expect(await collateral.status()).to.equal(CollateralStatus.DISABLED)
+    })
+
+    it('does revenue hiding correctly', async () => {
```

Comment:
this might be a good one to add to the generic file, with a switch it/it.skip switch to toggle it per suite

Reply:
Good idea. I tried to make it work for 30 minutes but couldn't because the Flux Finance fixture uses the real fTokens and not mocks, which makes it different enough from the rest of the collateral tests. Lot of new things would be required. I'm going to kick this one further down the road.